daemon/restore.go

```diff
@@ -0,0 +1,59 @@
+package daemon
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/docker/docker/container"
+	"github.com/docker/docker/runconfig"
+)
+
+// ContainerRestore restores the process in a container with CRIU
+func (daemon *Daemon) ContainerRestore(name string, opts *runconfig.CriuConfig, forceRestore bool) error {
+	container, err := daemon.Get(name)
+	if err != nil {
+		return err
+	}
+
+	if !forceRestore {
+		// TODO: It's possible we only want to bypass the checkpointed check,
+		// I'm not sure how this will work if the container is already running
+		if container.IsRunning() {
+			return fmt.Errorf("Container %s already running", name)
+		}
+
+		if !container.IsCheckpointed() {
+			return fmt.Errorf("Container %s is not checkpointed", name)
+		}
+	} else {
```

Comment:
Yes, I was thinking about the `force` case, but let's not do it atm then. I was not thinking about the case where you would do `docker commit` after `docker checkpoint` and then restore on top of the committed one. The ID would be different then.